src/quantization/sq4_quantizer.h

```diff
@@ -179,29 +198,33 @@ void
 SQ4Quantizer<metric>::ProcessQueryImpl(const DataType* query,
                                        Computer<SQ4Quantizer>& computer) const {
     try {
-        computer.buf_ = reinterpret_cast<uint8_t*>(this->allocator_->Allocate(this->code_size_));
-        this->EncodeOneImpl(query, computer.buf_);
+        computer.buf_ =
+            reinterpret_cast<uint8_t*>(this->allocator_->Allocate(this->dim_ * sizeof(float)));
```

Comment:
Should support  `int8` in future?

Reply:
we'll introduce different datatype as template params in future